hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockNodeConnectionManager.java

```diff
@@ -131,30 +131,90 @@ private synchronized void disconnectFromNode(@NonNull BlockNodeConfig node) {
         }
     }
 
-    private void streamBlockToConnections(@NonNull BlockState block) {
-        long blockNumber = block.blockNumber();
-        // Get currently active connections
-        List<BlockNodeConnection> connectionsToStream;
-        synchronized (connectionLock) {
-            connectionsToStream = activeConnections.values().stream()
-                    .filter(BlockNodeConnection::isActive)
-                    .toList();
-        }
+    public void streamBlockToConnections(@NonNull BlockState block) {
+        streamToConnections(
+                this::getActiveConnections,
+                () -> createPublishStreamRequests(block),
+                block.blockNumber(),
+                "block items");
+    }
 
+    public void streamBlockHeaderToConnections(long blockNumber, @NonNull Bytes blockHeader) {
+        streamToConnections(
+                this::getActiveConnections,
+                () -> createPublishStreamRequests(blockHeader),
+                blockNumber,
+                "block header");
+    }
+
+    /**
+     * Common streaming logic for both blocks and block headers.
+     *
+     * @param activeConnectionsSupplier supplier for the active block node connections to stream to
+     * @param blockNumber the block number
+     * @param requestsSupplier supplier for the requests to stream to block node connections
+     * @param contentType description of what's being streamed (for logging); empty string if whole block is streamed
+     */
+    private void streamToConnections(
+            @NonNull final Supplier<List<BlockNodeConnection>> activeConnectionsSupplier,
+            @NonNull final Supplier<List<PublishStreamRequest>> requestsSupplier,
```

Comment:
We should never have more than one active connection at a time. The first param can be
```
@NonNull final Supplier<BlockNodeConnection> activeConnectionSupplier
```